test/data_processing/test_data_processing.py

```diff
@@ -384,3 +387,7 @@ def test_averaging_and_svd(self):
         expected_std = np.array([np.std([1, 1, 1, -1, 1, 1, 1, -1]) / np.sqrt(8.0)] * 2)
         self.assertTrue(np.allclose(processed, np.array([0.5, -0.5]) / np.sqrt(2.0)))
         self.assertTrue(np.allclose(error, expected_std))
+
+
+if __name__ == "__main__":
+    unittest.main()
```

Comment:
Do we need this? Up to now I didn't need this when running tests locally and CI seemed fine without it.

Reply:
@eggerdj If I want to run a single test file, I usually write:
```
python <file name>
```
and it works, if these two lines appear at the end of the test file. I could of course, instead, specify the test that I want in a flag to `stestr` or `python -m unittest`.

Is there a reason to prefer to omit these lines?  @mtreinish

Reply:
I prefer to omit them, but I am happy to hear guidance from @mtreinish or @scasagrande.

Reply:
It's not really needed with modern test runners since they pretty much all let you run by file path natively. This is only really necessary with the stdlib unittest runner which doesn't not provide a useful UI that has any test selection. It doesn't harm anything by being there but I would say running with a more sophisticated test runner is in your best interest. Personally I never add this since most people don't manually invoke the stdlib runner and especially with larger test suites its just easier to use a single entrypoint from the runner that also gives more useful output.

For example, if you use pytest it would just be `pytest $path` or with stestr like we use via tox (which is also used in CI) you can do `stestr run -n $path`. The tox configuration is setup so you can just do `tox -epy -- -n $path` so you can call stestr natively this way in the tox venv.

Reply:
I typically use `pytest <path>` as well, so my inclination would be to drop it.